### PR TITLE
feat(indicator): add `<SIndicator>` component

### PR DESCRIPTION
**Pull request by kiaking**
I've changed props a bit from the issue.

```vue
<SIndicator state="running" />
```

```ts
interface Props {
  size?: Size
  state: State
  mode?: Mode
}

type Size =
  | 'nano'
  | 'mini'
  | 'small'
  | 'medium'
  | 'large'
  | 'jumbo'

type State =
  | 'pending'
  | 'ready'
  | 'queued'
  | 'running'
  | 'completed'
  | 'failed'

type Mode =
  | 'colored'
  | 'monochrome'
```

Screenshot 2024-06-03 at 17 50 03

### CODE REVIEW

File: lib/components/SIndicator.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+.SIndicator.large  { width: 40px; height: 40px; }
+.SIndicator.jumbo  { width: 48px; height: 48px; }
+
+.SIndicator.queued {
+  animation: indicator-blink 1s linear infinite;
+}
+
+.SIndicator.running {
+  animation: indicator-spin 1s linear infinite;
+}
```
Comment (brc-dd) [inline lib/components/SIndicator.vue:75]: maybe increase their duration a bit, they look too fast currently.

**Review by brc-dd — APPROVED** [review]
rest looks good

File: lib/components/SIndicator.vue (hunk 1 above)
Comment (brc-dd) [reply lib/components/SIndicator.vue:75]: `1400ms cubic-bezier(0.45, 0.05, 0.55, 0.95) infinite` looks better (keep .running linear)

Comment (kiaking) [reply lib/components/SIndicator.vue]: Updated to 1.5s, since we use 0.25, 0.5, 0.75 scales for animations 👀  And added that bezier. Yeah it looks nicer 👍

**Review by brc-dd — APPROVED** [review]

**Pull request merged**